### DISCUSSION TITLE
supabase.auth.resend method doesn't work with pkce?

### DISCUSSION THREAD

**Discussion by astonfuture** [supabase/supabase#18442] [General]
When resending email sign up confirmations with with the supabase.auth.resend() method, the token generated and placed in the email confirmation link isn't a pkce token so it fails to log the user in when using the pkce auth flow.

The confirming email part works but it doesn't log the user in automatically. Is this the desired behaviour? I would have thought it should behave the same as the confirmation email sent by the supabase.auth.signUp() method which does generate a pkce token.

**Comment by JS-GitRepo** [comment]
Hi @astonfuture , I'm experiencing the same issue currently and was wondering if you ever figured this out more definitively? Tried to create a serverClient specifying auth: {authflow: "pkce"} and can't seem to get that working with PKCE either.

**Reply by astonfuture** [reply]
IIRC, I ended up implementing a hack workaround where you simply try to sign the user up again with the provided details and that will trigger a resend of the confirmation email.

Haven't looked into this for a while so things may have changed with the updated versions of the packages. If you're using the latest version of the `supabase/ssr` package the auth flow _should_ be `pkce` by default.

**Reply by J0** [reply]
Hey!

Currently, Resend doesn't work with the PKCE Flow. This may change in the future as we're currently looking into improving the DX but it's not supported for now.

**Reply by JS-GitRepo** [reply]
Thank you both for the replies! Understood

**Reply by Coinners** [reply]
@J0 please work on this, i dont understand how you can recommend the pkce flow when it seems like it isnt even fully implemented

**Comment by Coinners** [comment]
Triggering Send magic link via UI on the users page also sends normal tokens
